lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -130,6 +130,14 @@ func (d *zfs) CreateVolume(vol Volume, filler *VolumeFiller, op *operations.Oper
 				if err != nil {
 					return err
 				}
+
+				// Move the GPT alt header to end of disk if needed.
+				err = moveGPTAltHeader(devPath)
+				if err == nil {
+					d.logger.Debug("Moved GPT alternative header to end of disk")
+				} else if err != nil && err != errNonGPTDisk {
+					return err
+				}
```

Comment:
So I'm not completely sure about that part, do we really want to do it in every driver whenever we alter the storage size (also need to deal with growing the volume) or can we maybe do this from the backend in one central place (all VM instance creations & when updating the size of a VM).